src/walletConnect/saga.ts

```diff
@@ -331,12 +347,58 @@ function getSupportedChains() {
   })
 }
 
+export function* normalizeTransaction(rawTx: any, network: Network) {
+  const tx: TransactionRequest = { ...rawTx }
+
+  // Handle `gasLimit` as a misnomer for `gas`, it usually comes through in hex format
+  if ('gasLimit' in tx && tx.gas === undefined) {
+    tx.gas = isHex(tx.gasLimit) ? hexToBigInt(tx.gasLimit as Hex) : (tx.gasLimit as bigint)
```

Comment:
why do we need conversion to a `BigInt`?

Reply:
good question, i moved this function from the request.ts file and didn't look at the contents too closely. since gasLimit shouldn't really exist, it's hard to anticipate what type it will be which is why i am assuming that the type cast existed in the first place. i have added a proper transform to bigint if the gasLimit is an acceptable type